sdk/projects/azure-projects/azure/projects/_component.py

```diff
@@ -0,0 +1,596 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import sys
+import keyword
+import types
+from collections import ChainMap
+from typing import (
+    Dict,
+    Generic,
+    Mapping,
+    Protocol,
+    Any,
+    Tuple,
+    Union,
+    Literal,
+    Optional,
+    List,
+    Type,
+    get_args,
+    get_origin,
+)
+from typing_extensions import Self, TypeVar, dataclass_transform
```

Comment:
TypeVar should be importable from base typing?

Reply:
If you import from `typing_extensions` you can use the new `default` feature that was introduced to base typing in Python 3.13 :)